tests/suites/manual/deploy_manual.sh

```diff
@@ -0,0 +1,202 @@
+    fi
+    set_verbosity
+}
+
+run_deploy_manual_lxd() {
+    echo
+
+    name="tests-$(petname)"
+
+    ssh-keygen -f "${TEST_DIR}/${name}" \
+        -t rsa \
+        -C "ubuntu@${name}.com" \
+        -N ""
+
+    create_priveleged_profile
+    create_user_profile "${name}"
+
+    series="bionic"
+
+    controller="${name}-controller"
+    model1="${name}-m1"
+    model2="${name}-m2"
+
+    launch_and_wait_addr() {
+        local container_name addr_result
+
+        container_name=${1}
+        addr_result=${2}
+
+        lxc launch --profile default \
+             --profile profile-privileged \
+             --profile "profile-${name}" \
+             ubuntu:"${series}" "${container_name}"
+
+        local address=""
+
+        attempt=0
+        while [ ${attempt} -lt 30 ]; do
+            address=$(lxc list "$1" --format json | \
+                jq --raw-output '.[0].state.network.eth0.addresses | map(select( .family == "inet")) | .[0].address')
+
+            if echo "${address}" | grep -q '^[0-9]\+\.[0-9]\+\.[0-9]\+\.[0-9]\+$'; then
+                echo "Using container address ${address}"
+                break
+            fi
+            sleep 1
+            attempt=$((attempt+1))
+        done
+
+        # shellcheck disable=SC2086
+        eval $addr_result="'${address}'"
+    }
+
+    launch_and_wait_addr "${controller}" addr_c
+    launch_and_wait_addr "${model1}" addr_m1
+    launch_and_wait_addr "${model2}" addr_m2
+
+    # shellcheck disable=SC2154
+    for addr in "${addr_c}" "${addr_m1}" "${addr_m2}"; do
+        ssh-keygen -f "${HOME}/.ssh/known_hosts" -R "${addr}"
+
+        attempt=0
+        while [ ${attempt} -lt 10 ]; do
+            OUT=$(ssh -T -n -i "${TEST_DIR}/${name}" \
+                -o IdentitiesOnly=yes \
+                -o StrictHostKeyChecking=no \
+                -o AddKeysToAgent=yes \
+                ubuntu@"${addr}" 2>&1 || true)
+            if echo "${OUT}" | grep -q -v "Could not resolve hostname"; then
+                echo "Adding ssh key to ${addr}"
+                break
+            fi
+
+            sleep 1
+            attempt=$((attempt+1))
+        done
+
+        if [ "${attempt}" -ge 10 ]; then
+            echo "Failed to add key to ${addr}"
+            exit 1
+        fi
+    done
+
+    cloud_name="cloud-${name}"
+
+    CLOUD=$(cat <<EOF
+clouds:
+  ${cloud_name}:
+    type: manual
+    endpoint: "ubuntu@${addr_c}"
+EOF
+)
+
+    echo "${CLOUD}" > "${TEST_DIR}/cloud_name.yaml"
+
+    juju add-cloud --client "${cloud_name}" "${TEST_DIR}/cloud_name.yaml" >"${TEST_DIR}/add-cloud.log" 2>&1
+
+    file="${TEST_DIR}/test-${name}.txt"
+
+    bootstrap "${cloud_name}" "test-${name}" "${file}"
+
+    juju add-machine ssh:ubuntu@"${addr_m1}" >"${TEST_DIR}/add-machine-1.log" 2>&1
+    juju add-machine ssh:ubuntu@"${addr_m2}" >"${TEST_DIR}/add-machine-2.log" 2>&1
+
+    juju enable-ha >"${TEST_DIR}/enable-ha.log" 2>&1
+
+    juju deploy percona-cluster
```

Comment:
Where does this get deployed?

Reply:
Unsure tbh, I'm going to revisit this next  available CI day.

Reply:
We want to add aws to this setup as well, so I'll generalize this when that happens